physics/discrete_trajectory_body.hpp

```diff
@@ -299,6 +382,18 @@ void DiscreteTrajectory<Frame>::WriteSubTreeToMessage(
     degrees_of_freedom.WriteToMessage(
         instantaneous_degrees_of_freedom->mutable_degrees_of_freedom());
   }
+  if (downsampling_.has_value()) {
```

Comment:
This code and the next function would be much cleaner if `Downsampling` had its own read/write methods.

Reply:
Done.